pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -996,21 +1046,17 @@ func (pl *dynamicResources) lookupClassParameters(logger klog.Logger, class *res
 		return parameters, nil
 	}
 
-	// TODO (https://github.com/kubernetes/kubernetes/issues/123731): use an indexer
-	allParameters, err := pl.classParametersLister.ResourceClassParameters(class.Namespace).List(labels.Everything())
+	indexerValue := fmt.Sprintf("%s/%s/%s/%s", class.ParametersRef.APIGroup, class.ParametersRef.Kind, class.ParametersRef.Namespace, class.ParametersRef.Name)
+	objs, err := pl.classParametersIndexer.ByIndex(classParametersGeneratedFromIndex, indexerValue)
 	if err != nil {
 		return nil, statusError(logger, fmt.Errorf("listing class parameters failed: %v", err))
 	}
-	for _, parameters := range allParameters {
-		if parameters.GeneratedFrom == nil {
-			continue
-		}
-		if parameters.GeneratedFrom.APIGroup == class.ParametersRef.APIGroup &&
-			parameters.GeneratedFrom.Kind == class.ParametersRef.Kind &&
-			parameters.GeneratedFrom.Name == class.ParametersRef.Name &&
-			parameters.GeneratedFrom.Namespace == class.ParametersRef.Namespace {
-			return parameters, nil
+	if len(objs) > 0 {
+		parameters, ok := objs[0].(*resourcev1alpha2.ResourceClassParameters)
```

Comment:
The previous code didn't check it because it would have caused additional work, but now it is cheap: can you check that `len(objs)` is exactly one? Use a `switch`, with one happy path and two different error paths (not found and too many).

Reply:
updated.

Reply:
/cc @pohly

Reply:
Forgot to ask earlier, sorry: can you add test scenarios which exercise the new code paths?

Reply:
@pohly added.

Reply:
Thanks!

/lgtm